browser/brave_ads/ads_service_factory.cc

```diff
@@ -106,4 +104,13 @@ bool AdsServiceFactory::ServiceIsNULLWhileTesting() const {
   return false;
 }
 
+std::unique_ptr<AdsTooltipsDelegateImpl>
+AdsServiceFactory::BraveAdsTooltipsDelegate(Profile* profile) const {
+#if BUILDFLAG(IS_ANDROID)
+  return nullptr;
+#else
+  return std::make_unique<AdsTooltipsDelegateImpl>(profile);
+#endif
+}
+
```

Comment:
Instead of doing it like this, I think it might be clearer to just have an Android-specific implementation of the tooltips delegate (`AdTooltipsDelegateAndroidImpl` maybe) that essentially does nothing on Android. Then, based on buildflag settings, just instantiate the appropriate delegate implementation.

Reply:
@emerick we need to pass nullptr here as we have a dependency on `AdsTooltipsController` with `AdsTooltipsDelegate`.